File: ANGULAR/src/app/models/product.ts
export class Product {
    productId: number;
    productName: string;
    productCatrgoryId: number;
    productCompanyId: number;
    productDescription: string;
    productPrice: number;
    productImage: string;
    productLastUpdate: Date;
    productAge: number;
    categoryName: string;
    companyName: string;  

    constructor(product: any) {
        this.productCatrgoryId = product.productCatrgoryId;
        this.productCompanyId = product.productCompanyId;
        this.productDescription = product.productDescription;
        this.productId = product.productId;
        this.productImage = product.productImage;
        this.productLastUpdate = product.productLastUpdate;
        this.productName = product.productName;
        this.productPrice = product.productPrice;
        this.productAge = product.productAge;
        this.categoryName = product.categoryName; 
        this.companyName = product.companyName
    }
}

File: C#/toys_shop/Dal/models/Product.cs
using System;
using System.Collections.Generic;

namespace Dal.models;

public partial class Product
{
    public int ProductId { get; set; }

    public string ProductName { get; set; } = null!;

    public int ProductCategoryId { get; set; }

    public int ProductCompanyId { get; set; }

    public string ProductDescription { get; set; } = null!;

    public int ProductPrice { get; set; }

    public string ProductImage { get; set; } = null!;

    public DateOnly ProductLastUpdate { get; set; }

    public virtual Category ProductCategory { get; set; } = null!;

    public virtual Company ProductCompany { get; set; } = null!;

    public virtual ICollection<PurchaseDetail> PurchaseDetails { get; set; } = new List<PurchaseDetail>();
}
